### ISSUE TITLE
Add Texture Splatting?

### ISSUE THREAD

**Issue by Thundros** [FarazzShaikh/Three.js-Procedural-Terrain#1]
Hi, 1st off, GREAT job on this project! PLEASE keep up the EXCELLENT work! :)

Second, can you add Texture Splatting with the ability to tile the textures with a chosen size?

Once again, keep up the GREAT work!

**Comment by FarazzShaikh** [comment]
of course! I'm actually working on that next, ill even try to implement vertex blending in order to achieve a better texture system

**Comment by Thundros** [comment]
@FarazzShaikh : WOW! Thank you SO much! That's AWESOME! Can you also add an export option with the ability to keep the awesome texture quality on the terrain?

**Comment by FarazzShaikh** [comment]
the flattening of the edges would make it a non-tillable texture if exported as a height map, i will implement a toggle for it. I'm still fairly new lol, ill surely work on the "Export as png" part now that I've added p5 support.

**Comment by Thundros** [comment]
@FarazzShaikh That's awesome! The main thing is being able to export the Terrain WITH the high tiled texture quality. To flatten the edges, I meant on the terrain mesh. Not on the PNG! :)

**Comment by Thundros** [comment]
Hello?

**Comment by FarazzShaikh** [comment]
yes i get what you're saying. sorry for the inconsistency, I have been busy, will be adding to this project when ever i get time :)

**Comment by Thundros** [comment]
when will you have time? i am very much interested in testing! :)

**Comment by FarazzShaikh** [comment]
hey, i finally had time to update the project, check it out ;)

**Comment by Thundros** [comment]
@FarazzShaikh : Cannot WAIT to see texture splatting / slope collision! <3 Merry Christmas! <3

**Comment by FarazzShaikh** [comment]
@Thundros I don't know a way of adding texture splatting yet. I'm working with the Three.js api and i was thinking more of a vertex shaded low poly look with vertex colors based on height. Collision too is a somewhat difficult to implement with three.js, however i will give it a shot.

**Comment by Thundros** [comment]
@FarazzShaikh : You use shaders from shadertoy.com :)

**Comment by Thundros** [comment]
@FarazzShaikh : & to save it with the new quality applied, you use a function which saves the new texture information along with the mesh information :)

**Comment by FarazzShaikh** [comment]
@Thundros couldn't get fragment shaders to work very well. I've added height based vertex shading instead. There's room to improve but the start look rather nice. ill be working on saving PBR maps from the simulation next.

**Comment by Thundros** [comment]
@FarazzShaikh : WONDERFUL! :) Keep up the great work! Also, there has GOT to be an easier shader to use that allows for excellent quality scaling of splat map textures.

**Comment by Thundros** [comment]
@FarazzShaikh : Any luck? :)
